yb-voyager/cmd/analyzeSchema.go

```diff
@@ -419,6 +413,138 @@ func checkStmtsUsingParser(sqlInfoArr []sqlInfo, fpath string, objType string) {
+	}
+
+	partitionTablesMap[fullyQualifiedName] = true // marking the partition tables in the map
+
+	var primaryKeyColumns, partitionColumns, uniqueKeyColumns []string
+
+	for _, column := range columns {
+		if column.GetColumnDef() != nil { //In case PRIMARY KEY constraint is added with column definition
+			constraints := column.GetColumnDef().Constraints
+			for _, constraint := range constraints {
+				if constraint.GetConstraint().Contype == pg_query.ConstrType_CONSTR_PRIMARY {
+					primaryKeyColumns = []string{column.GetColumnDef().Colname}
+				}
+				if constraint.GetConstraint().Contype == pg_query.ConstrType_CONSTR_UNIQUE {
+					uniqueKeyColumns = append(uniqueKeyColumns, column.GetColumnDef().Colname)
+				}
+			}
+		} else if column.GetConstraint() != nil {
+			//In case CREATE DDL has PRIMARY KEY(column_name) - it will be included in columns but won't have columnDef as its a constraint
+			for _, key := range column.GetConstraint().GetKeys() {
+				if column.GetConstraint().Contype == pg_query.ConstrType_CONSTR_PRIMARY {
+					primaryKeyColumns = append(primaryKeyColumns, key.GetString_().Sval)
+				} else if column.GetConstraint().Contype == pg_query.ConstrType_CONSTR_UNIQUE {
+					uniqueKeyColumns = append(uniqueKeyColumns, key.GetString_().Sval)
+				}
+			}
+		}
+	}
+
+	partitionElements := createTableNode.CreateStmt.GetPartspec().GetPartParams()
+
+	for _, partElem := range partitionElements {
+		if partElem.GetPartitionElem().GetExpr() != nil {
+			//Expression partitions
+			fmt.Printf("expression par on %s, %v", fullyQualifiedName, primaryKeyColumns)
+			if len(primaryKeyColumns) > 0 || len(uniqueKeyColumns) > 0 {
+				summaryMap["TABLE"].invalidCount[fullyQualifiedName] = true
+				reportCase(fpath, "Issue with Partition using Expression on a table which cannot contain Primary Key / Unique Key on any column",
+					"https://github.com/yugabyte/yb-voyager/issues/698", "Remove the Constriant from the table definition", "TABLE", fullyQualifiedName, sqlStmtInfo.formattedStmt, UNSUPPORTED_FEATURES, EXPRESSION_PARTIITON_DOC_LINK)
+				return
+			}
+		} else {
+			partitionColumns = append(partitionColumns, partElem.GetPartitionElem().GetName())
+		}
+	}
+
+	if len(partitionColumns) > 1 && createTableNode.CreateStmt.GetPartspec().GetStrategy() == pg_query.PartitionStrategy_PARTITION_STRATEGY_LIST {
+		summaryMap["TABLE"].invalidCount[fullyQualifiedName] = true
+		reportCase(fpath, `cannot use "list" partition strategy with more than one column`,
+			"https://github.com/yugabyte/yb-voyager/issues/699", "Make it a single column partition by list or choose other supported Partitioning methods", "TABLE", fullyQualifiedName, sqlStmtInfo.formattedStmt, UNSUPPORTED_FEATURES, LIST_PARTIION_MULTI_COLUMN_DOC_LINK)
+		return
+	}
+
+	if len(primaryKeyColumns) == 0 { // no need to report in case of non-PK tables
+		return
+	}
+
+	for _, pc := range partitionColumns {
+		if !slices.Contains(primaryKeyColumns, pc) {
```

Comment:
nit: let's do a set difference, and check if it's non-empty? 
Also, is there a way to add metadata to the unsupported feature? Would be ncie to report the missing columns as well.

Reply:
Added columns in analyze schema issue's reason as of now, not sure of assessment's unsupported feature.